File: examples/sample.cpp
// C++ sample
#include <iostream>
#include <string>
#include <vector>

struct Item {
    std::string name;
    double price;
};

double total(const std::vector<Item>& items) {
    double sum = 0.0;
    for (const auto& item : items) {
        sum += item.price;
    }
    return sum;
}

int main() {
    std::vector<Item> items{{"Keyboard", 79.5}, {"Mouse", 24.0}};
    std::cout << "Total: " << total(items) << '\n';
    return 0;
}

File: examples/sample.rs
// Rust sample
#[derive(Debug, Clone)]
struct Product {
    name: String,
    price: f64,
}

fn total(items: &[Product]) -> f64 {
    items.iter().map(|p| p.price).sum()
}

fn main() {
    let items = vec![
        Product { name: "SSD".into(), price: 129.0 },
        Product { name: "RAM".into(), price: 89.5 },
    ];

    let amount = total(&items);
    println!("total={amount:.2}, first={}", items[0].name);
}
